src/fstab-generator/fstab-generator.c

```diff
@@ -1239,13 +1370,31 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
                 } else
                         arg_volatile_mode = VOLATILE_YES;
 
-        } else if (streq(key, "systemd.swap")) {
+        } else if (streq(key, "systemd.mount")) {
 
-                r = value ? parse_boolean(value) : 1;
+                if (proc_cmdline_value_missing(key, value))
+                        return 0;
+
+                r = mount_array_add(value);
                 if (r < 0)
-                        log_warning("Failed to parse systemd.swap switch %s. Ignoring.", value);
-                else
-                        arg_swap_enabled = r;
+                        log_warning("Failed to parse systemd.mount= option, ignoring: %s", value);
+
+        } else if (streq(key, "systemd.swap")) {
+
+                if (!value)
+                        arg_swap_enabled = true;
+                else {
+                        r = parse_boolean(value);
+                        if (r >= 0)
+                                arg_swap_enabled = r;
+                        else {
+                                r = mount_array_add_swap(value);
+                                if (r < 0)
+                                        log_warning("Failed to parse systemd.swap= option, ignoring: %s", value);
+
+                                arg_swap_enabled = true;
+                        }
+                }
```

Comment:
i am not a fan of this overloading logic. i.e. that this both takes the boolean for disabling/enabling *and* the request to setup stuff. in particular as one is is additive, the other global.

Can we find a different name for the option? maybe systemd.mount-extra= systemd.swap-extra= or so, to avoid the confusion around the existing systemd.swap option?

Reply:
Using a short name like `systemd.swap=` would be nice, but I agree that the overload would be problematic. Esp. that it'd be common to omit options for swap, so we'd have an arbitrary argument with no separators or a boolean. Might be hard to distinguish the two in some cases.

I think it's OK to have a longer name. I don't think people will be using this often to specify file systems manually. We have root= and mount.usr= for the common cases, and the new options would most likely be created programatically by installers and such.